src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareConnectionProxy.java

```diff
@@ -674,6 +680,11 @@ protected synchronized void pickNewConnection() throws SQLException {
       return;
     }
 
+    if(Util.isNullOrEmpty(this.hosts)) {
```

Comment:
This block prevents calling failover() few line below. Failover() may update topology and successfully connect.

Reply:
If the host list is empty I don't think we want to fail over, we don't have any host/topology information so I don't think failover is possible (both failover procedures rely on being able to pull host information from the topology). And I don't think we can update topology because we don't use the current connection and we have no other information for hosts to connect to. Correct me if I'm wrong though